libs/data-display/src/DataTable.tsx

```diff
@@ -490,13 +490,32 @@ function DataTable<T extends object>({
   const primaryRow = findChild("DataTablePrimaryRow", children);
   const secondaryRow = findChild("DataTableSecondaryRow", children);
   const hasSecondaryColumns = React.isValidElement(secondaryRow);
-  const primaryColumnChildren = React.isValidElement(primaryRow) ? primaryRow.props.children : children;
-  const primaryColumnChildrenArray = React.Children.toArray(primaryColumnChildren).filter(Boolean);
+
+  const primaryColumnChildren = React.useMemo(() => {
+    return mapChildren(React.isValidElement(primaryRow) ? primaryRow.props.children : children);
+  }, [primaryRow, children]);
+
+  let primaryColumnChildrenArray = React.Children.toArray(primaryColumnChildren).filter(Boolean);
   const columnChildrenSize = primaryColumnChildrenArray.length;
 
-  const secondaryColumnChildren = React.isValidElement(secondaryRow) ? secondaryRow.props.children : null;
+  const secondaryColumnChildren = React.useMemo(() => {
+    return React.isValidElement(secondaryRow) ? mapChildren(secondaryRow.props.children) : null;
+  }, [secondaryRow, children]);
+
   const secondaryColumnChildrenArray = React.Children.toArray(secondaryColumnChildren).filter(Boolean);
 
+  function mapChildren(children: DataTableChild<T>[]) {
+    return children.map((child) => {
+      // @ts-ignore
```

Comment:
why all the ts-ignores here?

Reply:
Because it says that property children does not exist on type... which is true in most of the cases. But in cases like this when we send multiple optional columns than property children will exist in child.props as array and we need to extract those children. Anyway, Array.isArray() function should return false if we pass it undefined or null argument (or anything else that is not an array), so this ts-ignores should not cause any problems.